### PR TITLE
Add user documentation

### LINKED ISSUE #33: Add user documentation
Once we're stable we should do the whole sphinx + readthedocs kinda deal. Updating the README would also be nice.

### PR DESCRIPTION
**Pull request by Kyle-Verhoog**
A follow up PR will be needed to document each of the cli arguments and options but otherwise this should be a pretty solid starting point.

Resolves #33.

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -10,11 +10,17 @@
...
+copyright = "2020-Present Datadog, Inc"  # noqa
+
 extensions = [
-    "sphinx.ext.doctest",
     "sphinx.ext.autodoc",
+    "sphinx.ext.doctest",
+    "sphinx.ext.napoleon",
+    "sphinx_rtd_theme",
     "reno.sphinxext",
+    "sphinx_click",
```
Comment (jd) [inline docs/conf.py:23]: do you need to this something to `docs/requirements.txt`?

File: docs/quickstart.rst [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        venv = Venv(
+            pys=[3.9],
+            venvs=[
+                Venv(
+                    name="fmt",
+                    command="black .",
+                    pkgs={
+                        "black": "==20.8b1",
+                    },
+                ),
```
Comment (jd) [inline docs/quickstart.rst:23]: Reading this I'm wondering what happens if you do:

```python
venv = Venv(
            pys=[3.9],
            venvs=[
                Venv(
                    name="fmt",
                    pys=[3.8],
                    command="black .",
                    pkgs={
                        "black": "==20.8b1",
                    },
                ),
            ],
         )
```

with `pys` specified twice and in contradiction?

File: riot/riot.py [hunk 3]
```diff
@@ -63,6 +63,43 @@ def to_list(x: t.Union[_K, t.List[_K]]) -> t.List[_K]:
...
+              ),
+          ])
+
+    Args:
+        name (str): Name of the instance. Overrides parent value.
+        command (str): Command to run in the virtual environment. Overrides parent value.
+        pys  (List[float]): Python versions. Overrides parent value.
+        pkgs (Dict[str, Union[str, List[str]]]): Packages and version(s) to install into the virtual env. Merges and overrides parent values.
+        env  (Dict[str, Union[str, List[str]]]): Environment variables to define in the virtual env. Merges and overrides parent values.
+        venvs (List[Venv]): List of Venvs that inherit the properties of this Venv (unless they are overridden).
```
Comment (jd) [inline riot/riot.py:100]: It this valid rst/Sphinx format?
(first time I see it :))

Comment (Kyle-Verhoog) [reply riot/riot.py:100]: Yes! It's https://www.sphinx-doc.org/en/master/usage/extensions/napoleon.html

File: docs/quickstart.rst (hunk 2 above)
Comment (Kyle-Verhoog) [reply docs/quickstart.rst:23]: The child overrides the parent value so it would result in just `pys=[3.8]`

File: docs/conf.py (hunk 1 above)
Comment (Kyle-Verhoog) [reply docs/conf.py:23]: Yes, good catch!

**Review by majorgreys — APPROVED** [review]
docs look 👍

**Pull request merged**